src/map/entities/battleentity.cpp

```diff
@@ -733,7 +769,7 @@ void CBattleEntity::SetMLevel(uint8 mlvl)
 
 void CBattleEntity::SetSLevel(uint8 slvl)
 {
-    m_slvl = (slvl > (m_mlvl >> 1) ? (m_mlvl == 1 ? 1 : (m_mlvl >> 1)) : slvl);
+    m_slvl = (slvl > (m_mlvl * map_config.sj_level_ratio) ? (m_mlvl == 1 ? 1 : (m_mlvl * map_config.sj_level_ratio)) : slvl);
```

Comment:
I tagged you on a gist that shows you, I don't mind if you want to pr it (github should have sent you a notification, get me on irc if it didn't). There is also my old forum post. https://forums.dspt.info/viewtopic.php?f=20&t=1651&hilit=settings+setting#p8107

I did the same thing as you in my first version of it.